### PR TITLE
Remove built-in kubectl, filters, defaultNamespace, fix list bugs

### LINKED ISSUE #1003: Remove the built-in kubectl and replace it with plugin version in e2e tests
## Overview

Once the RBAC is added, we can finally remove the built-in kubectl and replace it with the plugin version. Here is a draft PR https://github.com/kubeshop/botkube/pull/986 that was closed as it was hard to maintain because of the changes on the `main` branch.

## Acceptance Criteria

- remove built-in `kubectl`
- use `kubectl` plugin in e2e tests
- probably adjusting messages, and adding some “hardcoded” ones directly in Botkube. For example, for the situation when kubectl is disabled - we had
   ```
   No `kubectl` commands are enabled in this channel. To learn how to enable them, visit https://docs.botkube.io/configuration/executor.
   ```
    and I cannot do that anymore directly in plugin, but we will be able to handle in Botkube Core, however new logic will need to be added.

### PR DESCRIPTION
**Pull request by mszostok**
## Description

Changes proposed in this pull request:

- Remove built-in kubectl, filters, defaultNamespace, fix list bugs

## Testing

Tested e2e.

## Related issue(s)

Fix #1003

### CODE REVIEW

**Comment by mszostok** [comment]
After rebase, I see that the name in the permission error is the one that we "hardcoded" on the backend. What should I do? Replace that in e2e test assertion? or maybe we should remove the hardcoded one and instead use the same name that was configured for group 🤔 ?

**Comment by pkosiec** [comment]
@mszostok yes, unfortunately K8s API returns just the username for authorization erorrs, without group name 😞 I think for tests we can use, just to make sure

Both options work for me (updating assertions or adding static user name for test cases), but I think the one with updating username in `e2e-test-values.yaml` for tests will be a bit better to debug if something goes wrong? 🤔

**Review by pkosiec — COMMENTED** [review]
LGTM, just one open question about help 👍

File: pkg/bot/interactive/help.go [hunk 1]
```diff
@@ -186,82 +173,31 @@ func (h *HelpMessage) configSections() []api.Section {
...
 			Base: api.Base{
-				Header:      "Run kubectl commands (if enabled)",
-				Description: fmt.Sprintf("You can run kubectl commands directly from %s!", formatx.ToTitle(h.platform)),
-			},
-			Buttons: []api.Button{
-				h.btnBuilder.ForCommandWithDescCmd(RunCommandName, "kubectl get services"),
-				h.btnBuilder.ForCommandWithDescCmd(RunCommandName, "kubectl get pods"),
-				h.btnBuilder.ForCommandWithDescCmd(RunCommandName, "kubectl get deployments"),
-			},
-		},
```
[the remark this answers is not in the document]
Comment (pkosiec) [reply pkg/bot/interactive/help.go:231]: Alright, fixed after f2f sync 👍 That was issue on my side.

**Review by pkosiec — APPROVED** [review]
LGTM

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -92,19 +92,34 @@ For faster development, you can also build and run Botkube outside K8s cluster.
...
+   communications:
+     default-group:
+       socketSlack:
+         enabled: true
+         channels:
+           default:
+             name: random
+         appToken: "xapp-xxxx"
+         botToken: "xoxb-xxxx"   
+   settings:
```
Comment (pkosiec) [inline CONTRIBUTING.md:108]:
```suggestion
   configWatcher:
      enabled: false
   settings:
```

**Review by pkosiec — APPROVED** [review]
👌 perfect!

**Pull request merged**